certification/internal/k8s/deployable_by_olm.go

```diff
@@ -0,0 +1,209 @@
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
+		return err
+	}
+
+	_, err = openshiftEngine.CreateCatalogSource(cli.CatalogSourceData{Name: catalogSource, Image: catalogImage}, cli.OpenShiftCliOptions{Namespace: catalogSourceNs}, k8sconfig)
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
+		return err
+	}
+
+	_, err = openshiftEngine.CreateOperatorGroup(cli.OperatorGroupData{Name: operatorGroup, TargetNamespaces: targetNamespaces}, cli.OpenShiftCliOptions{Namespace: ooNamespace}, k8sconfig)
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
+		return err
+	}
+
+	subscriptionData := cli.SubscriptionData{
+		Name:                   subscriptionName,
+		Channel:                channel,
+		CatalogSource:          catalogSource,
+		CatalogSourceNamespace: catalogSourceNs,
+		Package:                packageName,
+	}
+	_, err = openshiftEngine.CreateSubscription(subscriptionData, cli.OpenShiftCliOptions{Namespace: ooNamespace}, k8sconfig)
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
+		return err
+	}
+
+	return nil
+}
+
+func (p *DeployableByOlmCheck) cleanUp() {
+	log.Debug("Dumping data in artifacts/ directory")
+
+	subs, err := openshiftEngine.GetSubscription(subscriptionName, cli.OpenShiftCliOptions{Namespace: ooNamespace}, k8sconfig)
+	if err != nil {
+		log.Error("unable to retrieve the subscription")
+	}
+	p.writeToFile(subs, subscriptionName, "subscription")
+
+	cs, err := openshiftEngine.GetCatalogSource(catalogSource, cli.OpenShiftCliOptions{Namespace: catalogSourceNs}, k8sconfig)
+	if err != nil {
+		log.Error("unable to retrieve the catalogsource")
+	}
+	p.writeToFile(cs, catalogSource, "catalogsource")
+
+	og, err := openshiftEngine.GetOperatorGroup(operatorGroup, cli.OpenShiftCliOptions{Namespace: ooNamespace}, k8sconfig)
+	if err != nil {
+		log.Error("unable to retrieve the operatorgroup")
+	}
+	p.writeToFile(og, operatorGroup, "operatorgroup")
+
+	ns, err := openshiftEngine.GetNamespace(ooNamespace, k8sconfig)
+	if err != nil {
+		log.Error("unable to retrieve the namespace")
+	}
+	p.writeToFile(ns, ooNamespace, "namespace")
+
+	log.Trace("Deleting the resources created by Check")
+	openshiftEngine.DeleteSubscription(subscriptionName, cli.OpenShiftCliOptions{}, k8sconfig)
+	openshiftEngine.DeleteCatalogSource(catalogSource, cli.OpenShiftCliOptions{Namespace: catalogSourceNs}, k8sconfig)
+	openshiftEngine.DeleteOperatorGroup(operatorGroup, cli.OpenShiftCliOptions{}, k8sconfig)
+	openshiftEngine.DeleteNamespace(ooNamespace, cli.OpenShiftCliOptions{}, k8sconfig)
+}
+
+func (p *DeployableByOlmCheck) writeToFile(data interface{}, resource string, resourceType string) {
+	yamlData, err := yaml.Marshal(data)
+	if err != nil {
+		log.Error("unable to serialize the subscription")
+	}
+	err = ioutil.WriteFile(fmt.Sprintf("artifacts/%s-%s.yaml", resource, resourceType), yamlData, 0644)
+	if err != nil {
+		log.Error("failed to write the subscription object to the file")
+	}
+}
```

Comment:
The error messages indicate that we're only writing a subscription here. Is that the case?

Reply:
Yes. The intention is to dump all created resources in a file before the check exits, and disregard any error.